### DISCUSSION TITLE
Lesson 6: EIP-3855 warning on forge deploy script

### DISCUSSION THREAD

**Discussion by ironcladmerc** [Cyfrin/foundry-full-course-cu#988] [Q&A]
I'm getting this warning, that Patrick didn't get in the video, when I run the script to deploy SimpleStorage to Anvil:

```
EIP-3855 is not supported in one or more of the RPCs used.
Unsupported Chain IDs: 31337.
Contracts deployed with a Solidity version equal or higher than 0.8.20 might not work properly.
For more information, please see https://eips.ethereum.org/EIPS/eip-3855
```

I assume that's really nothing to worry about, but does that mean if we use solidity 0.8.20 it would not work with foundry? Does foundry need to add support for EIP-3855?

**Comment by shikhar229169** [comment]
Actually in the solidity versions 0.8.20 or higher there is a new opcode PUSH0 which is not supported on the Anvil chain, that's why it is just producing a warning.
But it is supported on sepolia chain, so you will get no warnings there.
I think that it will work with Anvil but it may create an issue.
So, just use the version 0.8.19 and there will be no issues.
